src/routes/usersRouter.js

```diff
@@ -1,9 +1,14 @@
 const express = require('express');
+
 const router = express.Router();
 
-const { updateUser, deleteUser } = require('../controller/userController');
+const {
+  updateUser,
+  deleteUser,
+  getUser,
+} = require('../controller/userController');
 
-//users == base path
-router.route('/:id').put(updateUser).delete(deleteUser);
+// users == base path
+router.route('/:id').put(updateUser).delete(deleteUser).get(getUser);
```

Comment:
Could we use `.patch` to update? It is more declarative because we would have a distinction in our code between creating and updating.

Reply:
why do you want to use patch?

Reply:
because then we will have patch for updating and post for creating. Remember to google before asking

PATCH
The HTTP PATCH method is defined in RFC 5789 as an extension to the earlier mentioned HTTP RFC. While PUT is used to replace an existing resource, PATCH is used to apply partial modifications to a resource.

Reply:
Also when Robert saw that Vasilis used post to update he made a comment on that.

Reply:
i googles before... just wanted to know your reasons.

Reply:
Oh, but those ones I said it here:

>  It is more declarative because we would have a distinction in our code between creating and updating.

Reply:
if we use patch : _"This means that you are only required to send the data that you want to update, and it won’t affect or change anything else. So if you want to update the first name on a database, you will only be required to send the first parameter; the first name."_

put: _"PUT is a method of modifying resource where the client sends data that updates the entire resource. It is used to set an entity’s information completely. PUT is similar to POST in that it can create resources, but it does so when there is a defined URI. PUT overwrites the entire entity if it already exists, and creates a new resource if it doesn’t exist.

For example, when you want to change the first name of a person in a database, you need to send the entire resource when making a PUT request."_

Reply:
i dont mind using patch i just want to understand why you prefer it .

Reply:
PUT can be for both, PATCH it is only update, so there is no wondering, if an engineer sees the code, know immediately   that that is an update.

Reply:
changed.

Reply:
> For example, when you want to change the first name of a person in a database, you need to send the entire resource when making a PUT request."_

Right and we don't want that.

Reply:
Could you see how it could be inefficient to send the whole item if you only need to change a field? And also how it makes the code more declarative and therefore helps to have specialized methods if we can? Or it doesn't make too much sense?

Reply:
it makes sense.